zerver/actions/streams.py

```diff
@@ -1300,6 +1300,51 @@ def do_change_stream_permission(
     )
 
 
+def get_users_string_with_permission(setting_value: int | AnonymousSettingGroupDict) -> str:
+    if isinstance(setting_value, int):
+        setting_group = NamedUserGroup.objects.get(id=setting_value)
+        return silent_mention_syntax_for_user_group(setting_group)
+
+    groups_with_permission = NamedUserGroup.objects.filter(id__in=setting_value.direct_subgroups)
+    group_names = [silent_mention_syntax_for_user_group(group) for group in groups_with_permission]
+
+    users_with_permission = UserProfile.objects.filter(id__in=setting_value.direct_members)
+    user_names = [silent_mention_syntax_for_user(user) for user in users_with_permission]
+
+    return ", ".join(group_names + user_names)
```

Comment:
`user_name_syntax_list` might be a better name here, since it is syntax wrapping the names?